`zkevm-sequencer` fail to start due to DB migration not finished in `zkevm-sync`

### Problem

When I run a full node using the `make run` command, it doesn't alway start all services correctly.

### Root cause analysis

```
# docker-compose ps
         Name                        Command                  State                                                                        Ports                                                                  
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
zkevm-aggregator          /bin/sh -c /app/zkevm-node ...   Up             0.0.0.0:50081->50081/tcp,:::50081->50081/tcp, 8123/tcp, 0.0.0.0:9093->9091/tcp,:::9093->9091/tcp                                        
zkevm-approve             /bin/sh -c /app/zkevm-node ...   Exit 0                                                                                                                                                 
zkevm-eth-tx-manager      /bin/sh -c /app/zkevm-node ...   Exit 1                                                                                                                                                 
zkevm-event-db            docker-entrypoint.sh postg ...   Up             0.0.0.0:5435->5432/tcp,:::5435->5432/tcp                                                                                                
zkevm-explorer-json-rpc   /bin/sh -c /app/zkevm-node ...   Up             8123/tcp, 0.0.0.0:8124->8124/tcp,:::8124->8124/tcp, 0.0.0.0:8134->8134/tcp,:::8134->8134/tcp                                            
zkevm-explorer-l1         /bin/sh -c mix do ecto.cre ...   Up             0.0.0.0:4000->4000/tcp,:::4000->4000/tcp                                                                                                
zkevm-explorer-l1-db      docker-entrypoint.sh postg ...   Up             0.0.0.0:5437->5432/tcp,:::5437->5432/tcp                                                                                                
zkevm-explorer-l2         /bin/sh -c mix do ecto.cre ...   Up             0.0.0.0:4001->4000/tcp,:::4001->4000/tcp                                                                                                
zkevm-explorer-l2-db      docker-entrypoint.sh postg ...   Up             0.0.0.0:5436->5432/tcp,:::5436->5432/tcp                                                                                                
zkevm-json-rpc            /bin/sh -c /app/zkevm-node ...   Up             0.0.0.0:8123->8123/tcp,:::8123->8123/tcp, 0.0.0.0:8133->8133/tcp,:::8133->8133/tcp, 0.0.0.0:9091->9091/tcp,:::9091->9091/tcp            
zkevm-l2gaspricer         /bin/sh -c /app/zkevm-node ...   Up             8123/tcp                                                                                                                                
zkevm-mock-l1-network     geth --http --http.api adm ...   Up             30303/tcp, 30303/udp, 0.0.0.0:8545->8545/tcp,:::8545->8545/tcp, 0.0.0.0:8546->8546/tcp,:::8546->8546/tcp                                
zkevm-pool-db             docker-entrypoint.sh postg ...   Up (healthy)   0.0.0.0:5433->5432/tcp,:::5433->5432/tcp                                                                                                
zkevm-prover              zkProver -c /usr/src/app/c ...   Up             0.0.0.0:50052->50052/tcp,:::50052->50052/tcp, 0.0.0.0:50061->50061/tcp,:::50061->50061/tcp, 0.0.0.0:50071->50071/tcp,:::50071->50071/tcp
zkevm-sequence-sender     /bin/sh -c /app/zkevm-node ...   Up             8123/tcp                                                                                                                                
zkevm-sequencer           /bin/sh -c /app/zkevm-node ...   Exit 1                                                                                                                                                 
zkevm-state-db            docker-entrypoint.sh postg ...   Up (healthy)   0.0.0.0:5432->5432/tcp,:::5432->5432/tcp                                                                                                
zkevm-sync                /bin/sh -c /app/zkevm-node ...   Up             8123/tcp 
```

I noticed that some containers such as `zkevm-sequencer` exited unexpectedly. By examining the logs:

```
# docker-compose logs --tail 80 zkevm-sequencer
zkevm-sequencer                | 2023-08-24T08:12:08.451Z       FATAL   cmd/run.go:261  error the component needs to run 6 migrations before starting. DB only contains 0 migrations
```

we see the problem is that some DB migration is not finished.

digging into the code, I found https://github.com/0xPolygonHermez/zkevm-node/blob/develop/cmd/run.go#L60. So it seems like the DB migration is handled by the synchronizer which is represented by the `zkevm-sync` container.

Then I looked into the Makefile and found:

```
run: ## Runs a full node
	...
	$(RUNSYNC)
	sleep 2
	$(RUNETHTXMANAGER)
	$(RUNSEQUENCER)
	...
```

Now it seems clear that `sleep 2` is the reason why sometimes I see the issue and sometimes I don't. It's likely sometimes the migration is not finished under 2 seconds.


### Solution

For now, we can just increase the time interval to make the issue less possible to happen as a workaround.

--- a/test/Makefile
+++ b/test/Makefile
@@ -292,9 +292,9 @@ stop-db: ## Stops the node database
 
 .PHONY: run-node
 run-node: ## Runs the node
-	$(RUNETHTXMANAGER)
 	$(RUNSYNC)
 	sleep 2
+	$(RUNETHTXMANAGER)
 	$(RUNSEQUENCER)
 	$(RUNSEQUENCESENDER)
 	$(RUNL2GASPRICER)

--- a/test/Makefile
+++ b/test/Makefile
@@ -293,7 +293,7 @@ stop-db: ## Stops the node database
 .PHONY: run-node
 run-node: ## Runs the node
 	$(RUNSYNC)
-	sleep 2
+	sleep 4
 	$(RUNETHTXMANAGER)
 	$(RUNSEQUENCER)
 	$(RUNSEQUENCESENDER)
@@ -463,7 +463,7 @@ run: ## Runs a full node
 	$(RUNAPPROVE)
 	sleep 3
 	$(RUNSYNC)
-	sleep 2
+	sleep 4
 	$(RUNETHTXMANAGER)
 	$(RUNSEQUENCER)
 	$(RUNSEQUENCESENDER)

